Use cursor.id for stable cursor navigation when not using orderBy

jonmagic opened the pull request:
This changes the connection query to use the primary key from the cursor (`cursor.id`) instead of the index (`cursor.index`) so that we can enjoy stable pagination for our use case (multiple people working a queue by dividing it up into smaller windows, each person removing entries from the queue for their window).

/cc @malandrina @hktouw @spicycode @erebor

malandrina commented on src/relay.js:
```diff
@@ -259,13 +260,21 @@ export function createConnectionResolver({
+          if (startIndex >= 0) options.offset = startIndex + 1;
+        } else {
+          let operator = args.after ?
+            (seqMajVer <= 3 ? '$gt' : Sequelize.Op.gt) : (seqMajVer <= 3 ? '$lt' : Sequelize.Op.lt)
```
This line is doing a lot! Could you extract some descriptively named variables?

```
const greaterThan = seqMajVer <= 3 ? '$gt' : Sequelize.Op.gt
const lessThan = seqMajVer <= 3 ? '$lt' : Sequelize.Op.lt
let operator = args.after ? greaterThan : lessThan
```

Part of me wonders if the knowledge of the version-specific operator should live in an object outside relay.js that already knows about `Sequelize`, but I'm not familiar enough with the code base to know what that object might be.

jonmagic replied:
The rest of this file uses similar nested ternary so I decided to look like everything else in here.

malandrina commented on src/base64.js:
```diff
@@ -1,9 +1,9 @@
 
 export function base64(i) {
-  return (new Buffer(i, 'ascii')).toString('base64');
+  return (new Buffer.from(i, 'ascii')).toString('base64');
```
Just curious, why is `Buffer.from` preferred?

spicycode replied:
> Just curious, why is `Buffer.from` preferred?

The old way of instantiating `Buffer` via the constructor (Node 6 and before) yielded very different behavior based on the argument types.  A few new methods (including `Buffer.from`) came to be in a more recent node version to help disambiguate things and make it N methods vs one method that accepts N argument types.

https://nodejs.org/api/buffer.html#buffer_buffer_from_buffer_alloc_and_buffer_allocunsafe has a decent short write-up on how things work now.

malandrina commented:
What are the odds we could get this merged into https://github.com/mickhansen/graphql-sequelize?

jonmagic commented:
> What are the odds we could get this merged into https://github.com/mickhansen/graphql-sequelize?

I addressed this in my work PR but basically I'm not ready yet. I would like to kick the tires on it in production and think about improvements before submitting a PR to the original project.

jonmagic commented:
@mickhansen I'm not ready to submit this to your project yet but wanted to make you aware of my work in this PR and see if you had any early feedback.

We're using `graphql-sequelize` (thank you for your project!!!) at GitHub on an internal project and the current pagination implementation doesn't quite work for us.

The internal app is a set of queues that a team of people work off of. Some of the queues are larger so we divide them between each team member by the date the queue entries were created.

With the current first/after and last/before query implementation it uses the offset value from the cursor rather than the primary key value from the cursor to query for the next set of entries from a queue. This breaks for us when we work an entry from the queue as we then have to subtract 1 from every offset.

This PR fixes the problem for us by changing the query to use the primary key from the cursor so we can do `WHEN id > ?` instead of `LIMIT N, OFFSET`.

It's pretty hacky the way I've implemented it now but I guess I want to ask if you would be open to a contribution that let you configure how those queries are performed? Btw, it's great that the cursors include the primary key AND the offset, so they can support either method of querying!

I've added a test in this PR that demonstrates the problem we encountered and that the (hacky) implementation makes the test pass.

Some posts I read while debugging our issues:

* https://www.novatec-gmbh.de/en/blog/art-pagination-offset-vs-value-based-paging/
* https://dev.to/jackmarchant/offset-and-cursor-pagination-explained-b89
* https://stackoverflow.com/questions/51795299/graphql-pagination-cursor-vs-offset
* https://engineering.mixmax.com/blog/api-paging-built-the-right-way

If you are open to a contribution along these lines I'll start working on a cleaner PR after we've proved out our implementation in production for a few weeks.
